src/debug_commands.h

```diff
@@ -17,3 +18,14 @@ typedef struct DebugCommandType {
 } DebugCommandType;
 
 int RegisterDebugCommands(RedisModuleCommand *debugCommand);
+
+typedef struct DebugCTX {
+  bool debugMode;
+  int maxDocsTBscanned;
+  int maxDocsTBscannedPause;
+  bool pauseBeforeScan;
+  volatile bool pause;
```

Comment:
Consider renaming `pause` and `pauseBeforeScan` to indicate they are related to BG indexing, or wrap all the BG indexing attributes in a BG indexing dedicated struct (even though today it's the only set of attributes)

Reply:
I was thinking that `pause` can be used by other flows as well for any pause-resume need.
I agree about the BG indexing struct.